### ISSUE TITLE
Calendar: Accessibility issue using calendar through NVDA screen readerComponent Name: Issue Title

### ISSUE THREAD

**Issue by nabil-elm** [primefaces/primevue#3877]
### Describe the bug

Here is my experience of using the calendar component by an NVDA version 2023.1 screen reader as blind user.
When focus is on the date combobox, the calendar popup automatically opens and NVDA announces:
"Choose Date  dialog, table, 19  row 5  We  column 4, clickable  table  with 7 rows and 7 columns  row 5  We  column 4  19, Choose Date, 19 Th  column 5  clickable,  20 Fr  column 6  clickable,  21 Sa  column 7  clickable,  22 row 6  Su  column 1  clickable, ..."

The following issues were encountered:
1. Automatically opening the calendar upon focus on input text field is not a very accessible approach. One of the WCAG guidelines pertains to predictability, which is crucial for ensuring a positive user experience for people with disabilities. Unexpectedly opening a dialog box can disrupt the user's interaction with the page and lead to confusion. As a blind person, I prefer to enter the date using the keyboard rather than the calendar when possible and have control for opening the calendar.
2. NVDA remains in "Browse Mode" and does not switch to "System Focus Mode" to maintain focus on the first selected date. As a result, NVDA continues to read the subsequent dates. To explore dates using keyboard arrows, I must manually switch NVDA's browse mode.
I believe the reason is that the focused element don't have an ARIA role: <span class="" draggable="false" aria-selected="false" tabindex="0">.
Two possible solutions are:
i. Make the parent node of <span> <td aria-label="15" class=""> the focusable element upon the calendar dialog opening.
ii. Assign the gridcell role to <span> and the none (or presentation) role to <td>.
3. In NVDA System Focus Mode, when I select date using the keyboard arrow key NVDA announces, for example, "15 row 4 Mo column 2, 15 section". I would prefer if NVDA did not repeat the day of the month followed by the section.
The issue arises from the definition of tabindex="0" on an element without a role. Assigning the gridcell role to <span> could solve this problem.
4. I have no way of knowing the month and year.
When navigating the calendar, NVDA only announces the day of the month, followed by row position, then the day of the week abbreviation, and finally the column position. For example: "12 row 4 We column 4, 12 section, 13 row 4 Th column 5, 13 section, ...".
It would be more accessible for me if:
A. NVDA announced the day, month, and year: "12 April 2023 row 4 Wednesday column 4". This can be achieved by using aria-label="12 April 2023" on <td>. It is better for me to know the selected date, especially when changing the month or year using keyboard shortcuts.
B. NVDA announced the full weekday name, not just the abbreviation. This can be achieved by using <th scope="col" aria-label="Wednesday" abbr="Wednesday">. Note: using abbr without aria-label does not make NVDA read Wednesday instead of We.
In the calendar component's accessibility specification, it states: "Calendar also includes a hidden section that is only available to screen readers with aria-live as "polite". This element is updated when the selected date changes to inform the user about the current date selected."
This solution is not implemented on the PrimeVue calendar demo page.
I think using aria-label to provide the date could offer a better user experience than using a hidden ARIA live polite section because NVDA will say "12 row 4 We column 4" before reading the hidden section. With aria-label, NVDA can say "12 April 2023 row 4 We column 4".
The date format I propose is not standard, but it presents date information according to its importance for a screen reader user.
5. The previous month button and the next month button do not have accessible names.
6. When I use the previous month button or the next month button, NVDA does not announce the current month and year.
In this case, using a hidden section that is only available to screen readers with aria-live as "polite" is a good way to provide month and year information to the screen reader.
7. The button to choose the month has the label "April" as both label and value, but the screen reader only reads "Choose month" (the value of aria-label). It would be better to have aria-label="April, choose month". The same remark is valid for the choice of the year.
A workaround is to remove aria-label="Choose month" and add aria-haspopup="dialog". NVDA will say "April button subMenu". The user will infer the purpose of the button.
8. When I activate the "Choose Month" button using NVDA's browse mode, NVDA places the focus on the first choice without switching to system focus mode.
The reason is the absence of an appropriate role in the span tag: <span class="p-monthpicker-month" tabindex="0">Jan ... .
A solution is to use role="menuitem" on the <span> tag and rolemenu on the <div>.
In general, it is recommended that elements focusable by tabindex="0" be interactive and have an ARIA role.
Other side, escape key doesn't close the choose month popup.

I appreciate the significant effort to make the components accessible. I hope these suggestions will improve the accessibility of the calendar component for NVDA users. Thank you for your attention to this matter.

### Reproducer

https://primevue.org/calendar

### PrimeVue version

3.27.0

### Vue version

3.x

### Language

TypeScript

### Build / Runtime

Vue CLI App

### Browser(s)

Chrome, firefox
